app/services/hyrax/characterization/valkyrie_characterization_service.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+require 'hydra-file_characterization'
+require 'nokogiri'
+
+class Hyrax::Characterization::ValkyrieCharacterizationService
+  ##
+  # @api public
+  # @param [Hyrax::FileMetadata] metadata  which has properties to recieve characterization values
+  # @param [Valkyrie::StorageAdapter::StreamFile] source to run characterization against
+  # @param [Hash] options the options pass to characterization
+  # @option options [Hash{Symbol => Symbol}] parser_mapping
+  # @option options [Class] parser_class
+  # @option options [Symbol] ch12n_tool
+  #
+  # @return [Hash]
+  def self.run(metadata, source = nil, **options)
+    new(metadata: metadata, source: source, **options).characterize
+    Hyrax.persister.save(resource: metadata)
+  end
+
+  ##
+  # @!attribute [rw] source
+  #   @return [Valkyrie::StorageAdapter::StreamFile]
+  # @!attribute [rw] metadata
+  #   @return [Hyrax::FileMetadata]
+  # @!attribute [rw] source
+  #   @return [Valkyrie::StorageAdapter::StreamFile]
+  # @!attribute [rw] tools
+  #   @todo clarify what this is for
+  #   @return [Symbol]
+  attr_accessor :mapping, :metadata, :parser_class, :source, :tools
+
+  ##
+  # @api private
+  def initialize(metadata:, # rubocop:disable Metrics/ParameterLists
+                 source:,
+                 characterizer: Hydra::FileCharacterization,
+                 parser_mapping: Hydra::Works::Characterization.mapper,
+                 parser_class: Hydra::Works::Characterization::FitsDocument,
+                 ch12n_tool: :fits)
+    @characterizer = characterizer
+    @metadata      = metadata
+    @source        = source
+    @mapping       = parser_mapping
+    @parser_class  = parser_class
+    @tools         = ch12n_tool
+  end
+
+  ##
+  # @api private
+  #
+  # Coerce given source into a type that can be passed to Hydra::FileCharacterization
+  # Use Hydra::FileCharacterization to extract metadata (an OM XML document)
+  # Get the terms (and their values) from the extracted metadata
+  # Assign the values of the terms to the properties of the metadata object
+  #
+  # @return [void]
+  def characterize
+    content = source_to_content
+    extracted_md = extract_metadata(content)
+    terms = parse_metadata(extracted_md)
+    store_metadata(terms)
+  end
+
+  protected
+
+  def source_to_content
+    source.rewind
+    source.read
+  end
+
+  def extract_metadata(content)
+    @characterizer.characterize(content, file_name, tools) do |cfg|
+      cfg[:fits] = Hydra::Derivatives.fits_path
+    end
+  end
+
+  def file_name
+    metadata.original_filename
+  end
+
+  # Use OM to parse metadata
```

Comment:
i don't understand how OM is involved.

it's deprecated since 2019, and isn't anywhere in the dependency tree. i think the reference is just confusing and can probably be removed?